packages/react/src/components/Accordion/__tests__/AccordionItem-test.js

```diff
@@ -13,9 +13,11 @@ import { render, screen } from '@testing-library/react';
 describe('AccordionItem', () => {
   describe('renders as expected - Component API', () => {
     it('should spread extra props onto outermost element', () => {
-      const { container } = render(<AccordionItem data-testid="test-id" />);
-
-      expect(container.firstChild).toHaveAttribute('data-testid', 'test-id');
+      render(<AccordionItem data-testid="test-id" />);
+      expect(screen.getByTestId('test-id')).toHaveAttribute(
+        'data-testid',
+        'test-id'
+      );
```

Comment:
Refactoring to use `screen.getByTestId` like this doesn't test to ensure the attribute is placed on the outermost element.

So this is a case in which we want to ignore the rule since we intentionally want to access the node to test that the data-testid is placed on the outermost element.

Reply:
I'm not understanding how `outermost` should work here, are you saying the parent component of `<AcordionItem />`?

i'm using debug with the old code and i'm getting the same result:


**container.firstChild**
```
    it('should spread extra props onto outermost element', () => {
      const { container, debug } = render(<AccordionItem data-testid="test-id" />);
      
      debug(container)

      expect(container.firstChild).toHaveAttribute('data-testid', 'test-id');
    })
    
 ```
 
![container_debug](https://user-images.githubusercontent.com/59191719/235540139-b3206d8c-a62c-4cb1-9baa-8fc4b9a8ca88.png)


**screen**
 
![screen_debug](https://user-images.githubusercontent.com/59191719/235540168-c1a58ea1-9614-46db-8b3e-b4d1c9385275.png)

Reply:
By "outermost" I mean the top-level parent element of the component. Our convention is that data-testid's should always be placed on the root/parent element of a component and not placed on inner children elements.

Given a simple component, such as

```js
const MyComponent = (props) => (
  <div data-testid={props['data-testid']}>
    <span>sample component</span>
  </div>
);
```

If `data-testid` was moved in this component source to no longer be on the outermost element

```js
const MyComponent = (props) => (
  <div>
    <span data-testid={props['data-testid']}>sample component</span>
  </div>
);
```

The test using `firstChild` would fail, as we want it to. The test using `screen.getByTestId` would still pass, which is not what we want.